mir_eval/segment.py

```diff
@@ -131,7 +132,7 @@ def validate_structure(reference_intervals, reference_labels,
         # Check only when intervals are non-empty
         if intervals.size > 0:
             # Make sure intervals start at 0
-            if not np.allclose(intervals[0, 0], 0.0):
+            if not np.allclose(intervals.min(), 0.0):
```

Comment:
This, and the max call below, could be made 2x faster as `intervals[:, 0].min()` and `intervals[:, 1].max()` respectively.